float32[] rawData
float32[] buffData